### PR TITLE
Filter builds according to modified generators

### PR DESCRIPTION
**Pull request by avdev4j**
We have some issues with Github action workflows at the moment, many of them are stuck in a queue. This is a Github bug, but I suggest to reduce the number of ran builds according to `paths` or `paths-ignore` properties to improve our CI.

WDYT @jhipster/developers ?

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
I like a lot, but I see 1 improvement here:
- add `default-app.yml` which will be triggered all the time -> maybe 1 angular, 1 react, 1 vue

**Comment by avdev4j** [comment]
I've added the default workflow. Tell me wdyt ;)

**Comment by avdev4j** [comment]
I've added a paths-ignore for default-app if we modify a client file. We need to discuss about that because it can be an issue:

Without paths-ignore:
- default-angular/react/vue will be launched twice if we do a modification in a specific client (e.g Angular).

With paths-ignore:
- default-angular/react will not be launched at all if we do a modification in a specific client (e.g Vue).

**Comment by mshima** [comment]
I don't like this change.
With this change, a PR like https://github.com/jhipster/generator-jhipster/pull/12164 won't trigger any build, but IMO it should trigger every build, webflux included which is not been triggered currently.

**Comment by mshima** [comment]
- Each e2e test tests a different combination of backend features.
- Push events should not be filtered IMO, there are not enough PR merges to worth it. A missed broken PR can be difficult to track.

**Comment by avdev4j** [comment]
> I don't like this change.
> With this change, a PR like #12164 won't trigger any build, but IMO it should trigger every build, webflux included which is not been triggered currently.

This is not really true. The default-app and the generator will be triggered, also all azure-pipelines ones, but I'm agree with that in certain cases we probably need to run all scenario. We should going further and find a solution.

Webflux workflow is now an option and it's not related to this PR.

File: .github/workflows/angular.yml [hunk 1]
```diff
@@ -22,9 +22,17 @@ on:
     push:
         branches-ignore:
             - 'dependabot/**'
+        paths:
+            - '!generators/client/**react.js'
+            - '!generators/client/**vue.js'
+            - 'generators/client/templates/angular/**'
```
Comment (mshima) [inline .github/workflows/angular.yml:28]: Drop filtering from push event.
```suggestion
```

File: .github/workflows/angular.yml [hunk 2]
```diff
@@ -22,9 +22,17 @@ on:
...
     pull_request:
         branches:
             - '*'
+        paths:
+            - '!generators/client/**react.js'
+            - '!generators/client/**vue.js'
+            - 'generators/client/templates/angular/**'
```
Comment (mshima) [inline .github/workflows/angular.yml:35]:
```suggestion
        paths:
            - 'utils/**'
            - 'test-integration/**'
            - 'generators/*.js'
            - '!generators/docker-*.js'
            - '!generators/*-docker.js'
            - '!generators/kubernetes-*.js'
            - 'generators/app/*.js'
            - 'generators/client/**/*.js'
            - '!generators/client/**/*-react.js'
            - '!generators/client/**/*-vue.js'
            - 'generators/client/templates/angular/**'
            - 'generators/server/**/*.js'
            - 'generators/server/templates/**'
            - 'generators/languages/**/*.js'
            - 'generators/languages/templates/**'
            - 'generators/common/**/*.js'
            - 'generators/common/templates/**'
            - 'generators/entity/**/*.js'
            - 'generators/entity-client/*.js'
            - 'generators/entity-client/templates/angular/**'
            - 'generators/entity-server/*.js'
            - 'generators/entity-server/templates/**
            - 'generators/entity-i18n/*.js'
            - 'generators/entity-i18n/templates/**
```

**Comment by avdev4j** [comment]
I think we can close this PR @mshima, can't we?

**Comment by mshima** [comment]
> I think we can close this PR @mshima, can't we?

Yep, If we became constrained in runners again we can create a new one, but with our current structure it's really complicated to filter by paths without loosing coverage.

**Comment by avdev4j** [comment]
great!

**Pull request closed without merging**